packages/@aws-cdk/aws-lambda-nodejs/test/function.test.ts

```diff
@@ -7,6 +7,10 @@ import { Stack } from '@aws-cdk/core';
 import { NodejsFunction } from '../lib';
 import { Bundling } from '../lib/bundling';
 
+const PROJECT_ROOT = process.env.NZM_SOURCE_PACKAGE
```

Comment:
Is there a way to avoid taking this dependency on the nozem
Environment?

Reply:
I don't think so. I'll be honest with you, the `aws-lambda-nodejs` package is the bane of my existence here.

In this case, this exists to tell it where the `.git` directory of the source repository is. I don't know why it needs to know that, but right now I'm just telling it in an attempt to get the build to pass.

Reply:
We lookup the project root because this is what we mount as `/asset-input` to be sure to have access to all node modules and dependencies when bundling. Would a fallback on looking for a lockfile (`yarn.lock`, `package-lock.json`) improve things here?

Reply:
I feel auto discovery of the project root is a source of confusion and odd issues for many users. Maybe we should just make that explicitly required?

Reply:
I think it's nice to have it auto discovered.

What about this?
```ts
// Find project root
const projectRoot = options.projectRoot
  ?? findUp(`.git${path.sep}`)
  ?? findUp('yarn.lock')
  ?? findUp('package-lock.json')
  ?? findUp('package.json');
```

here
https://github.com/aws/aws-cdk/blob/9ffb2682c167fe92e302bc322d60b9ae37de934a/packages/%40aws-cdk/aws-lambda-nodejs/lib/bundling.ts#L117-L121

Reply:
Using a lockfile in this case also doesn't help me. I'm trying to do an isolated build of the package. There's a `yarn.lock` somewhere but you shouldn't care about that.

My build script has set up the build environment exactly as Node-based tools should expect it (`node_modules` in the right place and everything). There's no `yarn.lock` or `package-lock.json` in sight.

Reply:
But there's a `package.json` right? so it should work.